lib/ansible/galaxy/token.py

```diff
@@ -54,11 +59,27 @@ def __init__(self, access_token=None, auth_url=None, validate_certs=True, client
         self.access_token = access_token
         self.auth_url = auth_url
         self._token = None
+        self._issued_at = None
+        self._expires_in = None
         self.validate_certs = validate_certs
         self.client_id = client_id
         if self.client_id is None:
             self.client_id = 'cloud-services'
 
+    @property
+    def expired(self):
+        if None in (self._issued_at, self._expires_in):
+            return False
+        try:
+            issue_time = datetime.strptime(self._issued_at, ISO_8601_FORMAT_STR)
+        except ValueError:
+            display.warning(f"KeycloakToken issued at unknown time format {self._issued_at}")
+            return False
+
+        # expire a few seconds before to handle proactively
```

Comment:
I saw this value being a whole minute in a GitHub bot project that had to work with GitHub API. The reasoning was that it'd also accommodate for the system clock being slightly out of sync with the actual time (when NTP detects such anomalies, it tends to correct time gently with multiple iterations instead of jumping the clock for relatively big amounts of time).

What is the typical expiration time for Galaxy tokens? If it's about an hour, 60 seconds might be a good trade-off.

Reply:
Found where I saw that: https://github.com/python-trio/snekomatic/blob/43e8e6d/snekomatic/gh.py#L278 / https://github.com/python-trio/snekomatic/blob/43e8e6d/snekomatic/gh.py#L86-L93.

Reply:
At least in the example I saw, it was 300 seconds, but I have yet to test/verify that's the current behavior. Guessing it's not an hour though if people are running into the token expiring when publishing a collection...

Reply:
Ah, fair enough. Maybe we need to tune it to correlate with the slow causes of HTTP request timeouts?

Reply:
The expiration for an AH token is 15 minutes, so I've changed this to refresh the token a minute early. Seem okay to you?

Reply:
Yep, that should be more than enough.